email_alert_service/models/unpublishing_alert.rb

```diff
@@ -0,0 +1,67 @@
+require "gds_api/email_alert_api"
+require "lib/uuid_v5"
+
+class UnpublishingAlert
+  def initialize(document, logger, unpublishing_scenario)
+    @document = document
+    @logger = logger
+    @unpublishing_scenario = unpublishing_scenario
+  end
+
+  def trigger
+    logger.info "Received unsubscription notification for #{document['title']}, unpublishing_scenario: #{unpublishing_scenario}, full payload: #{document}"
+    get_subscriber_list
+    bulk_unsubscribe
+  end
+
+private
+
+  attr_reader :document, :logger, :unpublishing_scenario, :content_item, :subscriber_list, :subscriber_list_id, :page_title
+
+  def bulk_unsubscribe
+    unless subscriber_list_id
+      logger.error "no subscription_list_id found for content id #{document['content_id']}"
```

Comment:
Since almost no content has  an associated single-page subscriber list, this is not an error.

Reply:
In fact, on line 49 you already log something for this case.

I think the logic would be clearer by removing this conditional and changing line 14 to `bulk_unsubscribe if subscriber_list_id`